File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        self.nodes = []

        self.appendNode(root)

        return self.nodes

    def appendNode(self, rootNode):
        if not rootNode:
            return
        
        self.appendNode(rootNode.left)
        self.nodes.append(rootNode.val)
        self.appendNode(rootNode.right)


        

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    std::vector<int> nodes;

public:
    vector<int> inorderTraversal(TreeNode* root) {
        append(root);

        return nodes;   
    }

private:
    void append(TreeNode* rootNode) {
        if (!rootNode)
            return;
        
        append(rootNode->left);
        nodes.push_back(rootNode->val);
        append(rootNode->right);
    }
};
